pygeobase/io_base.py

```diff
@@ -312,11 +443,55 @@ def _open(self, gpi):
         cell = self.grid.gpi2cell(gpi)
         filename = os.path.join(self.path, self.fn_format.format(cell))
 
-        if self.previous_cell != cell:
-            self.close()
+        if self.mode == 'r':
+            if self._read_bulk:
```

Comment:
This part of the implementation does not really enforce bulk reading. It just says "init a new ioclass with this filename". What the ioclass then does with it can not really be influenced here (other than the `ioclasskw`). It might read the whole file at once implementing `read_bulk` but it can also just read one `gpi` after the other. 

So since we can not exert any effective control over what we call `bulk` reading and writing at this level I would see the role of the `Gridded...` classes as just pointing the `ioclass` to the correct filename.

- In the case that the files are stored per grid point also the cell would have to change since the filename will have to change. At least in the current implementation.
- In the second use case I think that the `ioclass` has to handle the internal re-opening of the actual file.